src/Estimators/OneBodyDensityMatrices.cpp

```diff
@@ -85,6 +85,11 @@ OneBodyDensityMatrices::OneBodyDensityMatrices(OneBodyDensityMatricesInput&& obd
   if (is_spinor_)
     ssamples_.resize(samples_);
 
+  if (is_spinor_ && sampling_ != Sampling::METROPOLIS)
+    throw UniformCommunicateError("OneBodyDensityMatrices::OneBodyDensityMatrices only density sampling implemented "
```

Comment:
I still think throw from a constructor is a bad practice although I don't have a quick fix.